### ISSUE TITLE
error: 'OptionMap' in 'class folly::AsyncSocket' does not name a type

### ISSUE THREAD

**Issue by cello86** [facebook/folly#1522]
Hi All,
we tried to install mcrouter on a CentOS 7.x and we received this error related to folly:

```
In file included from ../../mcrouter/lib/network/CarbonMessageDispatcher.h:19:0,
                 from ../../mcrouter/lib/network/CarbonRequestHandler.h:13,
                 from ../../mcrouter/lib/network/gen/MemcacheServer.h:18,
                 from ../../mcrouter/lib/network/gen/Memcache.h:18,
                 from ../../mcrouter/lib/Reply.h:15,
                 from FailoverErrorsSettingsBase.h:15,
                 from FailoverErrorsSettingsBase.cpp:7:
../../mcrouter/lib/network/ConnectionOptions.h:46:45: error: 'OptionMap' in 'class folly::AsyncSocket' does not name a type
   using SocketOptions = folly::AsyncSocket::OptionMap;
```

Do we have to build something in folly to enable asyncSocket?

Thanks in advance,
Marcello

**Comment by Orvid** [comment]
This is likely the result of not matching the version of mcrouter to the version of Folly. Folly makes no API compatibility guarantees if you don't use the same tag to build Folly as you do the other project.

**Comment by cello86** [comment]
Hi @Orvid the mcrouter version is fixed as 0.40.0 generated at 04 November 2019 but in docker image generation script of mcrouter is not used the specific version but the latest master branch code.

**Comment by Orvid** [comment]
That would be a bug with mcrouter then, please file it against that repo.
